### PR TITLE
[RFR] Test to create service dialog from Tower job template

### PR DESCRIPTION
**Pull request by nachandr**
{{pytest: cfme/tests/infrastructure/test_config_management.py -v}}

## Purpose or Intent
Test to automate creation of service dialog from Ansible Tower job template
1)The existing TowerExplorerJobTemplateDetailsView was using an incorrect toolbar. Fixed that.
2)Created a new view for creation of service dailogs from Tower job templates.
2)Added a new test to create service dialog from Tower job template

### PRT results 
The test_config_system_tag[satellite_62] test  is failing on both 510 and 511 . The failure is not related to the code changes made through this PR.
All the other tests passed on both 510 and 511.

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit bd318c25b67dd907f75d487988b18a9a3cc30a8f:

`cfme/ansible_tower/explorer.py`:
- :red_circle: Line 217:12: **F821** *undefined name 'TowerExplorerJobServiceDialogView'*
- :bangbang: Line 221:101: **E501** *line too long (108 > 100 characters)*

`cfme/tests/infrastructure/test_config_management.py`:
- :red_circle: Line 176:91: **F821** *undefined name 'text'*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 89328cff37d43e724d91699ab3c2b9026aa1b605:

`cfme/ansible_tower/explorer.py`:
- :red_circle: Line 27:1: **F401** *'pytest' imported but unused*

`cfme/tests/infrastructure/test_config_management.py`:
- :red_circle: Line 176:91: **F821** *undefined name 'text'*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 7ab1111200f79ede51fc52e324b42fb6647f034d:

`cfme/tests/infrastructure/test_config_management.py`:
- :red_circle: Line 176:91: **F821** *undefined name 'text'*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/ansible_tower/explorer.py [hunk 1]
```diff
@@ -58,6 +59,12 @@ class TowerExplorerSystemJobTemplatesToolbar(View):
     view_selector = View.nested(ItemsToolBarViewSelector)
 
 
+class TowerExplorerSystemJobTemplatesDetailsToolbar(View):
+    reload = Button(title='Refresh this page')
```
Comment (digitronik) [inline cfme/ansible_tower/explorer.py]: Optional: `reload` was `build-in` function for py2 and moved to `importlib` in py3. still I would like to name such things with other name like `refresh` here. I know on same page its used multiple time. totally optional.

File: cfme/ansible_tower/explorer.py [hunk 2]
```diff
@@ -126,6 +133,20 @@ def is_displayed(self):
...
 
 
+class OptionForm(View):
+    dialog_name = TextInput('dialog_name')
+
+
+class TowerExplorerJobServiceDialogView(TowerExplorerView):
+    options = View.nested(OptionForm)
+    save_button = Button('Save')
+
```
Comment (digitronik) [inline cfme/ansible_tower/explorer.py]: Q: may be we will need `cancel` button as well? I won't found cancel button form `TowerExplorerView`.

File: cfme/ansible_tower/explorer.py [hunk 3]
```diff
@@ -126,6 +133,20 @@ def is_displayed(self):
...
+    @property
+    def is_displayed(self):
+        expected_title = 'Adding a new Service Dialog from "{}"'.format(self.context['object'].name)
+        return self.title.text == expected_title
```
Comment (digitronik) [inline cfme/ansible_tower/explorer.py]: I would also check `self.in_tower_explorer`... more checks in `is_displayed` always good as our navigation totally depends upon this property.

File: cfme/tests/infrastructure/test_config_management.py [hunk 4]
```diff
@@ -151,3 +169,27 @@ def test_ansible_tower_job_templates_tag(request, config_manager, tag):
...
+        assignee: nachandr
+        initialEstimate: 1/4h
+        casecomponent: Ansible
+        caseimportance: high
+
+    """
+    try:
+        job_template = config_manager.appliance.collections.ansible_tower_job_templates.all()[0]
+    except IndexError:
+        pytest.skip("No job template was found")
```
Comment (digitronik) [inline cfme/tests/infrastructure/test_config_management.py]: It look like `setup` means job_template required for this test... 
optional: you can move this as fixture.

File: cfme/tests/infrastructure/test_config_management.py [hunk 5]
```diff
@@ -24,11 +25,25 @@ def config_system(config_manager):
     return fauxfactory.gen_choice(config_manager.systems)
 
 
+@pytest.fixture
+def ansible_tower_dialog_through_ui(appliance):
+    """Returns service dialog object."""
+    service_dialogs = appliance.collections.service_dialogs
+    service_dialog = service_dialogs.instantiate(
```
Comment (digitronik) [inline cfme/tests/infrastructure/test_config_management.py]: Q: why are instantiate dialog without creating it? I think we need to create it here.

**Comment by dajoRH** [comment]
Lint report for commit 71ba38208ff764ad6dbb5c929f998a9007c8461d:

`cfme/tests/infrastructure/test_config_management.py`:
- :red_circle: Line 6:1: **F401** *'cfme.utils.appliance.implementations.ui.navigate_to' imported but unused*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Pull request merged**